app/controllers/employees/kudos_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Employees
+  class KudosController < EmployeesController
+    def index
+      render :index, locals: { kudos: Kudo.includes(%i[giver receiver company_value]).all }
+    end
+
+    def new
+      @kudo = Kudo.new
+      if current_employee.number_of_available_kudos <= 0
+        flash[:alert] = 'You have used all your kudos'
+        redirect_to root_path
+      else
+        render 'employees/kudos/new'
+      end
+    end
+
+    def edit
+      render :edit, locals: { kudo: Kudo.find(params[:id]) }
+    end
+
+    def create
+      @kudo = Kudo.new(kudo_params)
+      @kudo.giver_id = current_employee.id
+      Employee.transaction do
+        @kudo.save!
+        @current_employee.number_of_available_kudos -= 1
+        @current_employee.save!
+
+        flash[:notice] = 'Kudo was created successfully'
+        redirect_to root_path
+      rescue ActiveRecord::RecordInvalid
+        render 'employees/kudos/new'
+      end
+    end
+
+    def update
+      kudo = Kudo.find(params[:id])
+      if kudo.update(kudo_params)
+        flash[:notice] = 'Kudo was edited successfully'
+        redirect_to root_path
+      else
+        render :edit, locals: { kudo: }
+      end
+    end
+
+    def destroy
+      @kudo = Kudo.find(params[:id])
+      @kudo.destroy
+      flash[:notice] = if @kudo.destroy
```

Comment:
you call .destroy twice ; p. That's not that popular to check for success here. I'd use `.destroy!` and eventually use rescue_from in the controller above. "Eventually" because destroy is an action that we fully control, and it should work, once it doesn't, we (dev team) should be informed.